### PR TITLE
feat: Add snippets for VScode

### PR DESCRIPTION
**Pull request by c-vetter**

### CODE REVIEW

**Review by ploeh — COMMENTED** [review]
So, let me get this straight... Just by placing the snippets file in the (otherwise Git-ignored) `.vscode` directory, the code snippets light up for that Git repository?

lotr-lord-of-the-rings

**Comment by c-vetter** [comment]
> So, let me get this straight... Just by placing the snippets file in the (otherwise Git-ignored) `.vscode` directory, the code snippets light up for that Git repository?

That is correct. The `.vscode` directory is used for repository-local overrides and additions. Another neat feature is that you can recommend extensions for contributors with a `.vscode/extensions.json`. Although there's no good reason to edit that one manually, because that's what "Add to Workspace Recommendations" is for in the extension settings. Just `git add` it and it's good to go.

File: .vscode/contribution.code-snippets [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		]
+		"description": [
+			"Adds a single comment but not the surrounding comment section. \n"
+			"Use snippet \"addcommentsection\" for that."
+		]
+	}
+	"Add quote": {
+		"prefix": "addquote"
+		"body": [
+			"<figure class=\"quote\">"
```
[the remark this answers is not in the document]
Comment (c-vetter) [reply .vscode/contribution.code-snippets]: That is actually one of those things that get little attention, like accessibility. Pragmatically speaking, this is so far gone that it doesn't change much in terms of user interaction, if anything. But I'm a stickler for trying to be as possible to semantically correct as possible.

The `figure` element ties its children together as a semantic unit while allowing the `blockquote` to stay true to its intended semantics. The main point is that `blockquote` (and `q`) should only contain the quote itself while the source is considered meta data. Using the `footer` element-approaches that goal by clearly separating its content from the main part, which is why your pattern is a common one – even [MDN uses it](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/blockquote), unfortunately.

On another note, `cite` is expressly meant to contain a reference to a document. That's why it excludes the author in the snippet.

There are several conflicting sources on this topic, but the best overview I found is here: https://css-tricks.com/quoting-in-html-quotations-citations-and-blockquotes/#aa-hey-what-about-the-figure-element
The whole article explores different patterns, and the comments include a bit of additional detail, but I linked directly to the main paragraph, as I see it.

**Comment by ploeh** [comment]
FWIW, I'm already enjoying that these snippets are just available in the code base. Thank you 👍

**Pull request merged**